history/taskana-simplehistory-provider/src/test/java/acceptance/events/DeleteHistoryEventsOnTaskDeletionAccTest.java

```diff
@@ -0,0 +1,173 @@
+  private SimpleHistoryServiceImpl historyService;
+
+  @BeforeEach
+  public void setUp() {
+
+    taskService = taskanaEngine.getTaskService();
+    historyService = getHistoryService();
+  }
+
+  @Test
+  @WithAccessId(user = "admin")
+  void should_deleteHistoryEvents_When_TaskIsDeleted_With_HistoryDeletionEnabled()
+      throws Exception {
+
+    final String taskid = "TKI:000000000000000000000000000000000036";
+    taskanaEngineConfiguration.setDeleteHistoryOnTaskDeletionEnabled(true);
+
+    HistoryQueryMapper historyQueryMapper = getHistoryQueryMapper();
+
+    List<HistoryEventImpl> listEvents =
+        historyQueryMapper.queryHistoryEvent(
+            (HistoryQueryImpl) historyService.createHistoryQuery().taskIdIn(taskid));
+    assertThat(listEvents).hasSize(2);
+
+    taskService.deleteTask(taskid);
+
+    // make sure the task got deleted
+    ThrowingCallable getDeletedTaskCall =
+        () -> {
+          taskService.getTask(taskid);
+        };
+
+    assertThatThrownBy(getDeletedTaskCall).isInstanceOf(TaskNotFoundException.class);
+
+    listEvents =
+        historyQueryMapper.queryHistoryEvent(
+            (HistoryQueryImpl) historyService.createHistoryQuery().taskIdIn(taskid));
+    assertThat(listEvents).hasSize(0);
+  }
+
+  @Test
+  @WithAccessId(user = "admin")
+  void should_deleteHistoryEvents_When_TasksAreDeleted_With_HistoryDeletionEnabled()
+      throws Exception {
+
+    final String taskId_1 = "TKI:000000000000000000000000000000000037";
+    final String taskId_2 = "TKI:000000000000000000000000000000000038";
+
+    taskanaEngineConfiguration.setDeleteHistoryOnTaskDeletionEnabled(true);
+
+    HistoryQueryMapper historyQueryMapper = getHistoryQueryMapper();
+
+    List<HistoryEventImpl> listEvents =
+        historyQueryMapper.queryHistoryEvent(
+            (HistoryQueryImpl) historyService.createHistoryQuery().taskIdIn(taskId_1, taskId_2));
+    assertThat(listEvents).hasSize(3);
+
+    taskService.deleteTasks(Arrays.asList(taskId_1, taskId_2));
+
+    // make sure the tasks got deleted
+    ThrowingCallable getDeletedTaskCall =
+        () -> {
+          taskService.getTask(taskId_1);
+        };
+    ThrowingCallable getDeletedTaskCall2 =
+        () -> {
+          taskService.getTask(taskId_2);
+        };
+
+    assertThatThrownBy(getDeletedTaskCall).isInstanceOf(TaskNotFoundException.class);
+    assertThatThrownBy(getDeletedTaskCall2).isInstanceOf(TaskNotFoundException.class);
+
+    listEvents =
+        historyQueryMapper.queryHistoryEvent(
+            (HistoryQueryImpl) historyService.createHistoryQuery().taskIdIn(taskId_1, taskId_2));
+    assertThat(listEvents).hasSize(0);
+  }
+
+  @Test
+  @WithAccessId(user = "admin")
+  void should_notDeleteHistoryEvents_When_TaskIsDeleted_With_HistoryDeletionDisabled()
+      throws Exception {
+
+    final String taskId = "TKI:000000000000000000000000000000000039";
+
+    taskanaEngineConfiguration.setDeleteHistoryOnTaskDeletionEnabled(false);
```

Comment:
why are you doing this here? this belongs to `should_deleteHistoryEvents_When_TasksAreDeleted_With_HistoryDeletionEnabled` and should be disabled after that test (in order to not make the tests dependent from each other)